### PR TITLE
NamespaceMetadata Partitioning Alignment

### PR DESCRIPTION
**Pull request by shawnhathaway**
**What changed?**
- Rename NamespaceMetadata Shard to Partition
- Don't default NamespaceMetadata Partition in Schema
- Add ptn_id to Namespaces

**Why?**
Necessary for Vitess sharding support

**How did you test it?**
Unit testing

**Potential risks**
None

### CODE REVIEW

File: common/persistence/sql/sqlplugin/mysql/namespace.go [hunk 1]
```diff
@@ -33,38 +33,38 @@ import (
...
 	createNamespaceQuery = `INSERT INTO 
- namespaces (id, name, is_global, data, data_encoding, notification_version)
- VALUES(?, ?, ?, ?, ?, ?)`
+ namespaces (ptn_id, id, name, is_global, data, data_encoding, notification_version)
+ VALUES(?, ?, ?, ?, ?, ?, ?)`
 
 	updateNamespaceQuery = `UPDATE namespaces 
  SET name = ?, data = ?, data_encoding = ?, notification_version = ?
- WHERE shard_id=54321 AND id = ?`
+ WHERE ptn_id=54321 AND id = ?`
```
[the remark this answers is not in the document]
Comment (mastermanu) [reply common/persistence/sql/sqlplugin/mysql/namespace.go:41]: also interesting that create takes in partition ID as a parameter, but then get / update / lock assume a given partition ID, but don't fully understand this code

File: schema/mysql/v57/temporal/schema.sql [hunk 2]
```diff
@@ -1,20 +1,22 @@
 CREATE TABLE namespaces(
-  shard_id INT NOT NULL DEFAULT 54321,
+  ptn_id INT NOT NULL,
```
[the remark this answers is not in the document]
Comment (shawnhathaway) [reply schema/mysql/v57/temporal/schema.sql:2]: No particular reason, this is an unimportant column and figured the abbreviation was more convenient.

File: common/persistence/sql/sqlplugin/mysql/namespace.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shawnhathaway) [reply common/persistence/sql/sqlplugin/mysql/namespace.go:41]: Yea, it is oddly setup, cleaning this up is something to follow-up with definitely.

File: common/persistence/sql/sqlplugin/mysql/namespace.go [hunk 3]
```diff
@@ -33,38 +33,38 @@ import (
...
-	lockNamespaceMetadataQuery   = `SELECT notification_version FROM namespace_metadata FOR UPDATE`
-	updateNamespaceMetadataQuery = `UPDATE namespace_metadata SET notification_version = ? WHERE notification_version = ?`
+	getNamespaceMetadataQuery    = `SELECT notification_version FROM namespace_metadata WHERE partition_id = 54321`
+	lockNamespaceMetadataQuery   = `SELECT notification_version FROM namespace_metadata WHERE partition_id = 54321 FOR UPDATE`
+	updateNamespaceMetadataQuery = `UPDATE namespace_metadata SET notification_version = ? WHERE notification_version = ? AND partition_id = 54321`
 )
 
 const (
-	shardID = 54321
+	partitionId = 54321
```
Comment (alexshtin) [inline common/persistence/sql/sqlplugin/mysql/namespace.go:60]: nit:
```suggestion
	partitionID = 54321
```

Comment (alexshtin) [reply common/persistence/sql/sqlplugin/mysql/namespace.go:60]: Where does this number come from? I see it was default value in schama before.

Comment (shawnhathaway) [reply common/persistence/sql/sqlplugin/mysql/namespace.go]: Nowhere other than schema/current usage, just a constant value that is used to pin to a DB shard

**Pull request merged**